### ISSUE TITLE
Including two versions of the same package with the same .orig.tar.* fails and mucks up references.db

### ISSUE THREAD

**Issue by SwooshyCueb** [ionos-cloud/reprepro#9]
If I try to `includedsc` two source packages into the same component and distribution, and these two source packages include the original source, and the original source is the same for both (like if the debian version changed), the following error occurs:
```
BDB0696 Duplicate data items are not supported with sorted data
Internal error of the underlying BerkeleyDB database:
Within references.db subtable references at put: BDB0067 DB_KEYEXIST: Key/data pair already exists
```

Once it reaches this point, it has already added entries for the `dsc` and `debian.tar.*` files to `resources.db`. These changes are not reversed upon encountering the error, so the state of the repo is inconsistent.

**Comment by bdrung** [comment]
I can verify that behavior.

**Comment by bdrung** [comment]
Fixed and found another bug in the removesrc command that I will fix.

**Comment by bdrung** [comment]
Fixed in the [5.2.0-multiple-versions](/profitbricks/reprepro/tree/5.2.0-multiple-versions) branch.

**Comment by umagmrit** [comment]
@bdrung I am using `5.2.0-multiple-versions` branch and facing this issue

```
BDB0696 Duplicate data items are not supported with sorted data
Internal error of the underlying BerkeleyDB database:
Within references.db subtable references at put: BDB0067 DB_KEYEXIST: Key/data pair already exists
There have been errors!
```
